packages/core/admin/admin/src/components/Layout/HeaderLayout.tsx

```diff
@@ -0,0 +1,213 @@
+            <Flex>
+              {navigationAction && <Box paddingRight={3}>{navigationAction}</Box>}
+              <Box>
+                <Typography variant="beta" as="h1" {...props}>
+                  {title}
+                </Typography>
+                {isSubtitleString ? (
+                  <Typography variant="pi" textColor="neutral600">
+                    {subtitle}
+                  </Typography>
+                ) : (
+                  subtitle
+                )}
+              </Box>
+              {secondaryAction ? <Box paddingLeft={4}>{secondaryAction}</Box> : null}
+            </Flex>
+            <Flex>{primaryAction ? <Box paddingLeft={2}>{primaryAction}</Box> : undefined}</Flex>
+          </Flex>
+        </Box>
+      );
+    }
+
+    return (
+      <Box
+        ref={ref}
+        paddingLeft={10}
+        paddingRight={10}
+        paddingBottom={8}
+        paddingTop={navigationAction ? 6 : 8}
+        background="neutral100"
+        data-strapi-header
+      >
+        {navigationAction ? <Box paddingBottom={2}>{navigationAction}</Box> : null}
+        <Flex justifyContent="space-between">
+          <Flex minWidth={0}>
+            <Typography as="h1" variant="alpha" {...props}>
+              {title}
+            </Typography>
+            {secondaryAction ? <Box paddingLeft={4}>{secondaryAction}</Box> : null}
+          </Flex>
+          {primaryAction}
+        </Flex>
+        {isSubtitleString ? (
+          <Typography variant="epsilon" textColor="neutral600" as="p">
+            {subtitle}
+          </Typography>
+        ) : (
+          subtitle
+        )}
+      </Box>
+    );
+  }
+);
+
+/**
+ * useElementOnScreen: hook that returns a ref to an element and a boolean indicating if the element is in the viewport.
+ */
+
+const useElementOnScreen = <TElement extends HTMLElement = HTMLElement>(
+  // eslint-disable-next-line
+  options?: IntersectionObserverInit
+): [containerRef: React.RefObject<TElement>, isVisible: boolean] => {
+  const containerRef = React.useRef<TElement>(null);
+
+  const [isVisible, setIsVisible] = React.useState<boolean>(true);
+
+  // eslint-disable-next-line
+  const callback: IntersectionObserverCallback = ([entry]) => {
+    setIsVisible(entry.isIntersecting);
+  };
+
+  React.useEffect(() => {
+    const containerEl = containerRef.current;
+    const observer = new IntersectionObserver(callback, options);
+
+    if (containerEl) {
+      observer.observe(containerRef.current);
+    }
+
+    return () => {
+      if (containerEl) {
+        observer.disconnect();
+      }
+    };
+  }, [containerRef, options]);
+
+  return [containerRef, isVisible];
+};
+
+/**
+ * useCallbackRef: a custom hook that converts a callback to a ref to avoid triggering re-renders when passed as a
+ * prop or avoid re-executing effects when passed as a dependency
+ */
+
+function useCallbackRef<T extends (...args: any[]) => any>(callback: T | undefined): T {
```

Comment:
this can be imported from the DS

Reply:
I was bit hesitant to do that as its not used at many places in CMS but just 1 place and also its not something specific to DS.

Reply:
I'd still do it, I think we do it already in the CMS? If not we must have the hook elsewhere 🤔